### PR TITLE
Printing

### LINKED ISSUE #167: Exponential behavior when parsing shared term
I have this kore file: https://github.com/radumereuta/k-light/blob/evenLighter/src/test/k/unparametric_examples/imp/imp-rules.kore.info (about 300kB - the result of parsing IMP rules, with lots of sharing in them).
Sometime between June 6 and July 7 there has been a change which makes kore-parser take a HUGE amount of time to run (stopped it after 10 minutes).

### PR DESCRIPTION
**Pull request by bmmoore**
Fixes  #167

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

File: src/main/haskell/kore/src/Text/IndentingPrinter.hs [hunk 1]
```diff
@@ -1,43 +1,42 @@
...
-printToString printer = showChain ""
+printToString printer = renderString simpleStream
     where
-    writerAction = printer
-    readerAction = execWriterT writerAction
-    showChain = runReader readerAction 0
+      doc = execWriter printer
+      simpleStream = layoutPretty (defaultLayoutOptions {layoutPageWidth = Unbounded}) doc
+
+docToString :: Doc ann -> String
```
Comment (virgil-serbanuta) [inline src/main/haskell/kore/src/Text/IndentingPrinter.hs]: I know that the other public functions here do not have documentation, but they were written before the current style existed. Could you document this one?

**Comment by bmmoore** [comment]
@virgil-serbanuta I added comments to the new function, and also noticed that the `IsString ShowS` instance was not used anymore during the first commit.

Can you add some comments on other functions?

**Comment by virgil-serbanuta** [comment]
Sure, I added them.

**Comment by bmmoore** [comment]
Ah, the build was failing over some warnings I missed by not building with `--pedantic`. I dropped an unused import in IndentingPrinter, and renamed a few shadowing arguments from `list` to `items`.

**Pull request merged**